### PR TITLE
[RFC] Slight code cleanup

### PR DESCRIPTION
**Pull request by ebkalderon**
* Use the more idiomatic `&[Entity]` instead of `Vec<Entity>` in function signatures.
* Use `where` clause to shorten function signatures.
* Remove `// Unit tests` comment requirement from `CONTRIBUTING.md` section on testing.
* Add `cargo run --example {example-name}` to `CONTRIBUTING.md` section on testing.
* Update unit test example from `timing.rs` (no longer in the top-level `amethyst` crate) to `state.rs`.
* Remove unused `tasks` module.
* Miscellaneous formatting changes.
* Use HTTPS where possible in API docs.

### CODE REVIEW

**Comment by ebkalderon** [comment]
@amethyst/engine-devs Any comments? Should be pretty minor.

File: src/engine/app.rs [hunk 1]
```diff
@@ -56,20 +56,22 @@ impl Application {
...
-        for engine_event in engine_events {
-            self.context.lock().unwrap().broadcaster.publish().with::<EngineEvent>(engine_event);
+        let events = self.context.lock().unwrap().poll_engine_events();
+        for e in events {
+            self.context.lock().unwrap().broadcaster.publish().with::<EngineEvent>(e);
         }
-        let events = self.context.lock().unwrap().broadcaster.poll();
-        self.states.handle_events(events, self.context.lock().unwrap().deref_mut());
+
+        let ents = self.context.lock().unwrap().broadcaster.poll();
```
Comment (Emilgardis) [inline src/engine/app.rs]: Should be `let events = ...`

Comment (ebkalderon) [inline src/engine/app.rs]: No, that was intentional. It's short for `entities`. For some reason, `broadcaster.poll()` returns a list of entities, not events, despite being passed into `handle_events()`. I don't think I reviewed the code when it was first committed, so I think a refactor of Broadcaster might be necessary.

**Comment by kvark** [comment]
Gonna look later today...

**Comment by csherratt** [comment]
LGTM

**Comment by kvark** [comment]
@ebkalderon good stuff!

**Comment by ebkalderon** [comment]
Alright, looks like this is okay to merge!

**Pull request merged**